replit_river/session.py

```diff
@@ -382,9 +384,11 @@ async def send_message(
             serviceName=service_name,
             procedureName=procedure_name,
         )
-        TraceContextTextMapPropagator().inject(
-            msg, None, TransportMessageTracingSetter()
-        )
+        if span:
+            with use_span(span):
```

Comment:
Unfortunately the Propagator requires the span to be in the context and you can't just pass it directly. Should be fine because there's no await point inside the `with`